integration/client/client_test.go

```diff
@@ -322,7 +330,34 @@ func TestImagePullAllPlatforms(t *testing.T) {
 	defer cancel()
 
 	cs := client.ContentStore()
-	img, err := client.Fetch(ctx, imagelist.Get(imagelist.Pause))
+
+	// Retry the fetch operation to handle network flakiness
+	var img images.Image
+	maxRetries := 3
+	for i := 0; i < maxRetries; i++ {
+		img, err = client.Fetch(ctx, imagelist.Get(imagelist.Pause))
+		if err == nil {
+			break
+		}
+
+		// Check if this is a network-related error that should be retried
+		errStr := err.Error()
+		if strings.Contains(errStr, "connection reset by peer") ||
+			strings.Contains(errStr, "connection refused") ||
+			strings.Contains(errStr, "timeout") ||
+			strings.Contains(errStr, "network is unreachable") ||
+			strings.Contains(errStr, "temporary failure") {
```

Comment:
These errors are OS specific and could be easier to test for via errno, e.g., `ECONNRESET`, `ECONNREFUSED`, `ETIMEOUT`, and `ENETUNREACH`. Not sure what "temporary failure" represents error-wise since nothing like it's documented in `man errno` (did you mean "Resource Temporarily Unavailable", i.e., `EAGAIN`?).